src/main/java/appeng/crafting/v2/CraftingContext.java

```diff
@@ -31,9 +31,15 @@ public final class CraftingContext {
     public BaseActionSource actionSource;
 
     /**
-     * A working copy of the AE system's item list used for modelling what happens as crafting requests get resolved
+     * A working copy of the AE system's item list used for modelling what happens as crafting requests get resolved.
+     * Only extract, inject into {@link CraftingContext#byproductsInventory}.
      */
     public final MECraftingInventory itemModel;
+    /**
+     * An initially blank inventory for keeping all crafting byproduct outputs in.
+     * Extract from here before extracting from {@link CraftingContext#itemModel}
```

Comment:
I guess an explanation/motivation would also go nice here, as to why it is necessary to separate byproducts